src/v/raft/consensus.cc

```diff
@@ -912,6 +912,10 @@ ss::future<std::error_code> consensus::replace_configuration(
 }
 
 ss::future<> consensus::start() {
+    return ss::try_with_gate(_bg, [this] { return do_start(); });
```

Comment:
Just a thought, but would it be simpler overall to enforce a sequencing between start and stop, such that start must complete before stop can begin?

Reply:
it would. We have to refactor the whole starting/shutdown sequence in redpanda
.

Reply:
> Just a thought, but would it be simpler overall to enforce a sequencing between start and stop, such that start must complete before stop can begin?

yes it would, good observation. as Michal pointed out its a lot of refactoring in the application.cc and services involved. I think there are some details of the complexity in the commit message.